src/test/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilterUnitTest.java

```diff
@@ -0,0 +1,427 @@
+        // check that predictions from resource are identical to local
+        AssertJUnit.assertArrayEquals("Predictions via loading predictor from resource is not identical to local file",
+                resourceYProba, predictYProbaSerial, 0.0);
+    }
+
+    @Test(groups = "sv")
+    protected void testFeatureConstruction() {
+        final XGBoostEvidenceFilter evidenceFilter = new XGBoostEvidenceFilter(
+                evidenceList.iterator(), readMetadata, params, emptyCrossingChecker
+        );
+        for(int ind = 0; ind < featuresTestData.stringReps.length; ++ind) {
+            final BreakpointEvidence evidence = evidenceList.get(ind);
+            final String stringRep = featuresTestData.stringReps[ind];
+            final EvidenceFeatures fVec = featuresTestData.features[ind];
+
+            final BreakpointEvidence convertedEvidence = BreakpointEvidence.fromStringRep(stringRep, readMetadata);
+            final String convertedRep = convertedEvidence.stringRep(readMetadata, params.minEvidenceMapQ);
+            AssertJUnit.assertEquals("BreakpointEvidence.fromStringRep does not invert BreakpointEvidence.stringRep",
+                    stringRep.trim(), convertedRep.trim());
+            final EvidenceFeatures calcFVec = evidenceFilter.getFeatures(evidence);
+            AssertJUnit.assertArrayEquals("Features calculated by XGBoostEvidenceFilter don't match expected features",
+                    fVec.getValues(), calcFVec.getValues(), featuresTol);
+        }
+    }
+
+    @Test(groups = "sv")
+    protected void testFilter() {
+        final List<BreakpointEvidence> expectedPassed = new ArrayList<>();
+        int index = 0;
+        for(final BreakpointEvidence evidence : evidenceList) {
+            final double proba = featuresTestData.proba[index];
+            if(proba > params.svEvidenceFilterThresholdProbability) {
+                expectedPassed.add(evidence);
+            }
+            index += 1;
+        }
+
+        final XGBoostEvidenceFilter evidenceFilter = new XGBoostEvidenceFilter(
+                evidenceList.iterator(), readMetadata, params, emptyCrossingChecker
+        );
+        final List<BreakpointEvidence> passedEvidence = new ArrayList<>();
+        evidenceFilter.forEachRemaining(passedEvidence::add);
+
+        AssertJUnit.assertEquals("Evidence passed by XGBoostEvidenceFilter not the same as expected",
+                expectedPassed, passedEvidence);
+    }
+
+    private static double[] predictProba(final Predictor predictor, final FVec[] testFeatures) {
+        final int numData = testFeatures.length;
+        final double[] yProba = new double[numData];
+        for(int rowIndex = 0; rowIndex < numData; ++rowIndex) {
+            yProba[rowIndex] = predictor.predictSingle(testFeatures[rowIndex], false,0);
+        }
+        return yProba;
+    }
+
+    static class JsonMatrixLoader {
```

Comment:
Is there not a Java library that can do a lot of this for you?

Reply:
I'm using the jackson JSON library to deal with the hard stuff. However I have to share tabular data from python in the form of pandas DataFrames and numpy arrays with Java. And I'm also passing around heterogenous data (string representations, probabilities, feature matrices). So a lot of this is just niche. Next major PR should be for a java training tool, in which case passing information should be much simpler (serialize / deserialize).